devices/flash-zynq7000/flashdrv.c

```diff
@@ -131,7 +145,8 @@ static int flashdrv_wipCheck(time_t timeout)
 	time_t start = hal_timerGet();
 
 	do {
-		if ((res = flashdrv_statusRegGet(&st)) < 0)
+		res = flashdrv_statusRegGet(&st);
+		if (res < 0)
 			return res;
 
 		if ((hal_timerGet() - start) >= timeout)
```

Comment:
out of this PR, but this will probable be better
```
	const time_t stop = hal_timerGet() + timeout;

	do {
		if ((res = flashdrv_statusRegGet(&st)) < 0)
		res = flashdrv_statusRegGet(&st);
		if (res < 0)
			return res;

		if (hal_timerGet() >= stop)
```
[same in](https://github.com/phoenix-rtos/phoenix-rtos-devices/blob/63235c8966ae99b274010a7762ea4916ad44b82d/storage/zynq7000-flash/flashdrv.c#L152-L166)

Reply:
Ok, I accept Your suggestion.